### ISSUE TITLE
Invalid CycloneDX JSON generated for specific images like "monetdb/monetdb:Jun2023-SP1" and "redis"

### ISSUE THREAD

**Issue by Parisha7** [anchore/syft#2268]
**What happened**:
We had observed an error when trying to attest an SBOM file containing an image "monetdb/monetdb:Jun2023-SP1".  
However, later after investigation, we found out that the generated SBOM file was at fault because of one section containing "python3-libs and python3" dependency. Snippet for the same section is attached in the last.

We also started getting error on the redis image and found out that error has started coming while upgrading from syft version `0.72.0`   to `0.74.1`. The reason is that the license part earlier contained `name` and now it is started coming as `id`. If we test it with updating `name` under license manually, then it is passing for both the images mentioned above.

Using [cyclonedx-cli](https://github.com/CycloneDX/cyclonedx-cli), I found out that it was invalid.

Unfortunately the tool does not give any more information that what was wrong in the SBOM file.
```
Unable to validate against any JSON schemas.
BOM is not valid.
```

**What you expected to happen**:
I would have expected the SBOM to be valid, no matter the input image.
We generate SBOMs on a large scale with Syft, and this kind of issues we are facing with multiple images.

**Steps to reproduce the issue**:
Generate the SBOM using below command and validate using cyclonedx-cli.
```
 syft -q redis:latest -o cyclonedx-json --file alpine_sbom14.json
 
 .\cyclonedx-win-x86.exe validate --input-file .\alpine_sbom14.json

```
**Anything else we need to know?**:
Unfortunately, I must not share the broken SBOM file with the outside world.

I look forward to your answers!

**Environment**:
- Output of `syft version`: `0.82.0` and tried with latest as well
- OS (e.g: `cat /etc/os-release` or similar): Linux

```
      {
         "bom-ref":"pkg:rpm/fedora/python3-libs@3.11.4-1.fc38?arch=x86_64&upstream=python3.11-3.11.4-1.fc38.src.rpm&distro=fedora-38&package-id=b4aeecc42347b0db",
         "type":"library",
         "publisher":"FedoraProject",
         "name":"python3-libs",
         "version":"3.11.4-1.fc38",
         "licenses":[
            {
               "license":{
                  "id":"Python-2.0.1"
               }
            }
         ],
         "cpe":"cpe:2.3:a:fedoraproject:python3-libs:3.11.4-1.fc38:*:*:*:*:*:*:*",
         "purl":"pkg:rpm/fedora/python3-libs@3.11.4-1.fc38?arch=x86_64&upstream=python3.11-3.11.4-1.fc38.src.rpm&distro=fedora-38"
      }
	  
	  
	        {
         "bom-ref":"pkg:rpm/fedora/python3@3.11.4-1.fc38?arch=x86_64&upstream=python3.11-3.11.4-1.fc38.src.rpm&distro=fedora-38&package-id=29226dff1289a55b",
         "type":"library",
         "publisher":"FedoraProject",
         "name":"python3",
         "version":"3.11.4-1.fc38",
         "licenses":[
            {
               "license":{
                  "id":"Python-2.0.1"
               }
            }
         ],
         "cpe":"cpe:2.3:a:fedoraproject:python3:3.11.4-1.fc38:*:*:*:*:*:*:*",
         "purl":"pkg:rpm/fedora/python3@3.11.4-1.fc38?arch=x86_64&upstream=python3.11-3.11.4-1.fc38.src.rpm&distro=fedora-38"
      },
```

**Comment by Parisha7** [comment]
Also, could you plz let us know if there is any verbose mode in syft tool commands that we can use. Thanks.

**Comment by spiffcs** [comment]
👋 Hey @Parisha7 are you able to update to the latest syft version? This is what I see on my end:
```
❯ syft version
Application: syft
Version:    0.94.0
BuildDate:  2023-10-20T17:21:07Z
GitCommit:  8f6bdde6662aa8050a71eadbdb7bd5a3b079a56d
GitDescription: v0.94.0
Platform:   darwin/amd64
GoVersion:  go1.21.3
Compiler:   gc
```

<img width="811" alt="Screenshot 2023-10-30 at 6 57 03 PM" src="https://github.com/anchore/syft/assets/32073428/bea24603-7902-443e-9004-e4fa6484fdfd">

In the above image `sbom-utility` is on version `v0.13.0`

The specification version is `CycloneDX 1.4`

**Comment by spiffcs** [comment]
Ahh I see where the disconnect is - I used https://github.com/CycloneDX/sbom-utility/releases/tag/v0.13.0

You're using [cyclonedx-cli](https://github.com/CycloneDX/cyclonedx-cli). Let me try the other one.

**Comment by spiffcs** [comment]
> The reason is that the license part earlier contained name and now it is started coming as id. If we test it with updating name under license manually, then it is passing for both the images mentioned above.

Are you sure this is an issue with the document? What does the schema field say? `  "$schema": "http://cyclonedx.org/schema/bom-1.4.schema.json",

ID is the correct field as of `cyclonedx-1.4`

https://cyclonedx.org/docs/1.4/json/#components_items_licenses_items_license_id

When using `cyclonedx-cli` I see an error thrown for the following component of the redis sbom:
```
./cyclonedx-osx-x64 validate --fail-on-errors --input-version v1_4 --input-file alpine_sbom14.json > error
```

```
Validating JSON BOM...
Validation failed: 
#/properties/components/items
On instance: #/components:
[
    {
      "bom-ref": "pkg:deb/debian/adduser@3.134?arch=all&distro=debian-12&package-id=5aeab6f311d0e8bb",
      "type": "library",
      "publisher": "Debian Adduser Developers <adduser@packages.debian.org>",
      "name": "adduser",
      "version": "3.134",
..................
```

Checking now to see what might be incorrect

**Comment by spiffcs** [comment]
No results on the above investigation - the tool does not provide a great resource for figuring out what it thinks is incorrect about the SBOM. 

So far:
- https://github.com/CycloneDX/sbom-utility/releases/tag/v0.13.0 says 🟢 
- https://github.com/CycloneDX/cyclonedx-cli 🔴 

The final test I did was I checked our own internal validations:
- https://github.com/anchore/syft/blob/main/schema/cyclonedx/Makefile

I changed the input to the redis image you provided:
```
element id: Schemas validity error : Element '{http://cyclonedx.org/schema/bom/1.5}id': [facet 'enumeration'] The value 'Kazlib' is not an element of the set
```

This seems to be an error on the schema of cyclonedx:
https://spdx.org/licenses/

Kazlib is an official spdx license ID. It also seems to be an oversight to include a hard `enum` here for schema versions given the frequency at which the list is updated.

We can issue a "fix" which removes a valid license ID from generated SBOM, but I think the upstream fix is the actual concern here where cyclonedx is treating a valid license ID as invalid.

**Comment by Parisha7** [comment]
Hi @spiffcs 

Is there any input pending on our side?

Can you provide us with an ETA for this issue, that would be really helpful. 

Thanks

**Comment by Parisha7** [comment]
Hi @spiffcs 

Any update on this issue.

**Comment by spiffcs** [comment]
@Parisha7 Like I said in the earlier comment - we're waiting for the cyclone-dx schema to reflect the latest version of the SPDX license list.

I just ran `syft -o cyclonedx dir:.github/` on a local github directory and did not see any of the invalid URL anymore - are you still seeing `git@github.com` links with the latest syft?

**Comment by Parisha7** [comment]
The issue is resolved with latest cyclonedx-cli version 0.25.1, therefore closing this ticket.

**Comment by Parisha7** [comment]
The issue is resolved with latest cyclonedx-cli version 0.25.1, therefore closing this ticket.
